01_fib/shuutech/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+)
+
+//var out io.Writer = os.Stdout
+var arrayList = []int{0, 1}
+
+func nextVal(a int, b int, n int, c int) []int {
+	numPair := []int{a, b}
+	var i = numPair[0] + numPair[1]
+	var j = numPair[1] + i
+	newPair := []int{i, j}
+	c++
+	if c < n {
+		arrayList = append(arrayList, newPair...)
+		nextVal(i, j, n, c)
```

Comment:
You are quite correct. I subsequently found that there are performance benefits to using a loop instead of recursive which I will keep in mind for next time. I specifically wanted to challenge myself to use recursive (as a way to get more practice using recursive)  https://medium.com/@felipedutratine/iterative-vs-recursive-vs-tail-recursive-in-golang-c196ca5fd489.

Reply:
:+1: ok, sound good